pages/team-structure.js

```diff
@@ -0,0 +1,402 @@
+    }
+
+    onDeterminePage = (e) => {
+        if ((this.state.page+1) * 4 >= this.props.roleCards.length) {
+            this.setState({page: 0})
+        } else {
+            this.setState({page: this.state.page + 1})
+        }
+    }
+
+    render() {
+        const {selectedTeam, tierId, roleCards, onSelect} = this.props;
+        if (selectedTeam !== undefined) {
+            if (roleCards.length > 4){
+                return(
+                <SystemComponent position="relative">
+                    <TeamHierMemberListGrid 
+                        tierId={tierId}
+                        roleCards={roleCards.slice(this.state.page * 4, Math.min( (this.state.page+1) * 4, roleCards.length))}
+                        currentPage={this.state.page}
+                        onSelect={(cardId, tierId) => onSelect(cardId, tierId)}>
+                    </TeamHierMemberListGrid>
+                    <NextSetButton onSelect={this.onDeterminePage}></NextSetButton>
+                </SystemComponent>)
+            } else {
+                return(
+                    <SystemComponent position="relative">
+                        <TeamHierMemberListGrid 
+                            tierId={tierId}
+                            roleCards={roleCards} 
+                            currentPage={this.state.page}
+                            onSelect={(cardId, tierId) => onSelect(cardId, tierId)}>
+                        </TeamHierMemberListGrid>
+                    </SystemComponent>
+                )
+            }
+        } else{
+            return(<React.Fragment></React.Fragment>)
+        }
+    }
+}
+
+
+class TeamHierarchy extends React.Component {
+
+    constructor(props){
+        super(props)
+        // NOTE: selectedMemberId -- each tier's card has a unique id
+        // selectedTier -- each tier has a unique ID
+        this.state = {
+            memberSelected: false,
+            selectedMemberId: undefined,
+            selectedTierId: 1,
+            selectedTeam: undefined,
+        };
+    }
+
+    onSelectMember = (cardId, tierId) => {
+        // TODO: put a condition ??
+        if (tierId === 2) {
+            const selectedSubTeamCard = subteam_json.filter(function(subTeamCard){return subTeamCard._id === cardId})[0]
+            this.setState({
+                selectedTierId: tierId, 
+                selectedMemberId: selectedSubTeamCard._id, 
+                memberSelected: true, 
+                selectedTeam: selectedSubTeamCard._id
+            })
+        } else {
+            this.setState({
+                selectedTierId: tierId, 
+                selectedMemberId: cardId, 
+                memberSelected: true, 
+                selectedTeam: undefined
+            })
+        }
+    }
+
+    render() {
+        return (
+            <PageTemplate title="Team Structure">
+                <React.Fragment>
+                <ThemeProvider theme={teamHierCustomTheme}>
+                    <SystemComponent 
```

Comment:
Can you use the Card component here? Also do we need the black border? It makes it inconsistent with the rest of the components.

Reply:
Removed the black border. One of the direct children of this SystemComponent you are referring to is already a Card component. Card applies a big padding, so I'm afraid that making this SystemComponent a Card as well will making the paddings add on top of each other and hard to deal with. I'll keep this parent component as a SystemComponent if that's okay.